src/telemetry.ts

```diff
@@ -0,0 +1,38 @@
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { BaseLanguageClient, ClientCapabilities, StaticFeature } from 'vscode-languageclient';
+
+const TELEMETRY_VERSION = 1;
+
+type TelemetryEvent = {
+  v: number;
+  name: string;
+  properties: { string };
+};
+
+export class TelemetryFeature implements StaticFeature {
+  constructor(private client: BaseLanguageClient, private reporter: TelemetryReporter) {
```

Comment:
`reporter` isn't used outside the constructor, so you might not need to create a class property for it.

If we want to keep it for later usage, we could use `this.reporter` in line 20.